src/server.c

```diff
@@ -4287,8 +4287,13 @@ int processCommand(client *c) {
         addReply(c,shared.queued);
     } else {
         int flags = CMD_CALL_FULL;
-        if (client_reprocessing_command) flags |= CMD_CALL_REPROCESSING;
+        if (client_reprocessing_command) {
+            flags |= CMD_CALL_REPROCESSING;
+            c->flags |= CLIENT_REPROCESSING_COMMAND;
```

Comment:
i think it would be better to set that flag inside call (based on the input flag).
this way if we someday call `call` from another place, it'll be handled.